hoomd/GSDDequeWriter.h

```diff
@@ -31,7 +31,7 @@ class PYBIND11_EXPORT GSDDequeWriter : public GSDDumpWriter
 
     void analyze(uint64_t timestep) override;
 
-    void dump();
+    void dump(long int start, long int end, bool clear_entire_buffer);
```

Comment:
Let's move `clear_entire_buffer` to a class level attribute (with get/set methods in C++ and a `param_dict` entry in Python). This way, users only need to set the parameter once and leave it.

I think the default could be either `True` or `False`. For backward compatibility, `dump()` with no arguments writes and clears all frames with either setting. We should set the default to be what we think is the most common use-case.